cli/src/CLIArgsParser.h

```diff
@@ -15,90 +15,26 @@
 * limitations under the License.
 ********************************************************************************/
 
-#ifndef OPENSPACENET_OPENSPACENETARGS_H
-#define OPENSPACENET_OPENSPACENETARGS_H
+#ifndef OPENSPACENET_CLIARGSPARSER_H
+#define OPENSPACENET_CLIARGSPARSER_H
 
+#include <OpenSpaceNetArgs.h>
+#include <OpenSpaceNet.h>
 #include <boost/program_options.hpp>
-#include <utility/Logging.h>
-#include <vector/Feature.h>
-
-#define OSN_LOG(sev) DG_LOG(OpenSpaceNet, sev)
-#define MAPSAPI_MAPID  "digitalglobe.nal0g75k"
-
 
 namespace dg { namespace osn {
 
-enum class Source
-{
-    UNKNOWN,
-    LOCAL,
-    DGCS,
-    EVWHS,
-    MAPS_API,
-    TILE_JSON
-};
-
-enum class Action
-{
-    UNKNOWN,
-    HELP,
-    DETECT,
-    LANDCOVER
-};
-
-class OpenSpaceNetArgs
+class CLIArgsParser
 {
 public:
-    // Input options
-    Action action = Action::UNKNOWN;
-    Source source = Source::UNKNOWN;
-
-    std::string image;
-    std::unique_ptr<cv::Rect2d> bbox;
-
-    // Web service input options
-    std::string token;
-    std::string credentials;
-    int zoom = 18;
-    int maxConnections = 10;
-    std::string mapId = MAPSAPI_MAPID;
-    std::string url;
-    bool useTiles=false;
-
-
-    // Output options
-    deepcore::vector::GeometryType geometryType = deepcore::vector::GeometryType::POLYGON;
-    std::string outputFormat = "shp";
-    std::string outputPath;
-    std::string layerName;
-    bool producerInfo = false;
-    bool append = false;
-
-    // Processing options
-    bool useCpu = false;
-    float maxUtilization = 95;
-    std::string modelPath;
-    std::unique_ptr<cv::Size> windowSize;
-
-    // Feature detection options
-    float confidence = 95;
-    std::unique_ptr<cv::Point> stepSize;
-    bool pyramid = false;
-    bool nms = false;
-    float overlap = 30;
-    std::vector<std::string> includeLabels;
-    std::vector<std::string> excludeLabels;
-    std::vector<int> pyramidWindowSizes;
-    std::vector<int> pyramidStepSizes;
-
-    // Logging options
-    bool quiet = false;
     dg::deepcore::level_t consoleLogLevel = dg::deepcore::level_t::info;
     std::string fileLogPath;
     dg::deepcore::level_t fileLogLevel = dg::deepcore::level_t::debug;
 
-    OpenSpaceNetArgs();
-    void parseArgsAndProcess(int argc, const char* const* argv);
+    CLIArgsParser();
+    void setupArgParsing(int argc, const char* const* argv);
+    void startOSNProcessing();
+    OpenSpaceNetArgs osnArgs;
```

Comment:
Actually, never mind. I see what you did there. We'll have to discuss it further, but disregard this comment.